libs/core/tests/unit_tests/output_parsers/test_pydantic_parser.py

```diff
@@ -96,3 +101,125 @@ def test_json_parser_chaining(
     assert res["f_or_c"] == "C"
     assert res["temperature"] == 20
     assert res["forecast"] == "Sunny"
+
+
+class Actions(Enum):
+    SEARCH = "Search"
+    CREATE = "Create"
+    UPDATE = "Update"
+    DELETE = "Delete"
+
+
+class TestModel(BaseModel):
+    action: Actions = Field(description="Action to be performed")
+    action_input: str = Field(description="Input to be used in the action")
+    additional_fields: Optional[str] = Field(
+        description="Additional fields", default=None
+    )
+    for_new_lines: str = Field(description="To be used to test newlines")
+
+
+# Prevent pytest from trying to run tests on TestModel
+TestModel.__test__ = False  # type: ignore[attr-defined]
+
+
+DEF_RESULT = """{
+    "action": "Update",
+    "action_input": "The PydanticOutputParser class is powerful",
+    "additional_fields": null,
+    "for_new_lines": "not_escape_newline:\n escape_newline: \\n"
+}"""
+
+# action 'update' with a lowercase 'u' to test schema validation failure.
+DEF_RESULT_FAIL = """{
+    "action": "update",
+    "action_input": "The PydanticOutputParser class is powerful",
+    "additional_fields": null
+}"""
+
+DEF_EXPECTED_RESULT = TestModel(
+    action=Actions.UPDATE,
+    action_input="The PydanticOutputParser class is powerful",
+    additional_fields=None,
+    for_new_lines="not_escape_newline:\n escape_newline: \n",
+)
+
+
+def test_pydantic_output_parser() -> None:
+    """Test PydanticOutputParser."""
+
+    pydantic_parser: PydanticOutputParser = PydanticOutputParser(
+        pydantic_object=TestModel
+    )
+
+    result = pydantic_parser.parse(DEF_RESULT)
+    print("parse_result:", result)  # noqa: T201
+    assert DEF_EXPECTED_RESULT == result
+    assert pydantic_parser.OutputType is TestModel
+
+
+def test_pydantic_output_parser_stream() -> None:
+    """Test PydanticOutputParser in a streaming sequence."""
+
+    class Person(BaseModel):
+        name: str
+        age: int
+
+    def llm(input: Any) -> Iterable[str]:
+        yield from [
+            "{",
+            '"name": "',
+            "King",
+            " Louis",
+            ' XVI"',
+            ', "age": 3',
+            "8}",
+        ]
+
+    parser = PydanticOutputParser(pydantic_object=Person)
+    chain = llm | parser
+    assert chain.invoke({}) == Person(name="King Louis XVI", age=38)
+    stream_result = list(chain.stream({}))
+    assert stream_result == [
+        Person(name="King Louis XVI", age=3),
+        Person(name="King Louis XVI", age=38),
+    ]
```

Comment:
why isn't name streamed

Reply:
because before there's any age, it doesn't comply to pydantic schema (requires int age)